main/src/main/java/org/mobilitydata/gtfsvalidator/validator/TripTravelSpeedValidator.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2021 MobilityData IO
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.validator;
+
+import com.google.common.collect.Multimaps;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map.Entry;
+import org.mobilitydata.gtfsvalidator.annotation.GtfsValidator;
+import org.mobilitydata.gtfsvalidator.annotation.Inject;
+import org.mobilitydata.gtfsvalidator.notice.FastTravelBetweenStopsNotice;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsStop;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTime;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTimeTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsTripTableContainer;
+import org.mobilitydata.gtfsvalidator.type.GtfsTime;
+import org.mobilitydata.gtfsvalidator.util.GeospatialUtil;
+
+/**
+ * Validates: all records of `trips.txt` refer to a collection of {@code GtfsStopTime} from file
+ * `stop_times.txt` of which -when sorted by `stop_sequence`- the travel speed between each stop is
+ * not above 150 km/h
+ *
+ * <p>Time complexity: <i>O(n * p)</i>, where <i>n</i> is the number of records in
```

Comment:
I believe this time complexity is wrong? If it's just measuring stop to stop distance and scanning through stop_times.txt once, then it should be O(n), where n is the number of records in stop_times.txt.

Reply:
Thanks for flagging this, I modified the javadocs. But it seems to me that the complexity would be O(n*p), where 
- n: the number of trips in `trips.txt`
- p: the number of stop times referred to by a trip. 
For each trip we calculate the speed between each pair of stop times.

Reply:
Well, yes, that's another way to phrase it but when possible I think it's more useful to give a tighter bound with a single metric based on a single file, especially in relationship to the files that end up being huge like stop_times.txt and shapes.txt. Each stop time can only belong to one trip, so you're only processing each stop_times.txt record once. And each speed measurement between pair of stop times is constant. So you can really forget about the trips table, it's only the size of stop_times.txt that matters for big O notation here.

Reply:
Alright, `stop_times.txt` takes precedence over `trips.txt` because of the size of the files!  Will modify :)